libflux/src/ast/walk/mod.rs

```diff
@@ -0,0 +1,363 @@
+#[cfg(test)]
+mod tests;
+
+use crate::ast::*;
+use std::cell::{RefCell, RefMut};
+use std::rc::Rc;
+
+// Node represents any structure that can appear in the AST
+#[derive(Debug)]
+pub enum Node<'a> {
+    Package(&'a Package),
+    File(&'a File),
+    PackageClause(&'a PackageClause),
+    ImportDeclaration(&'a ImportDeclaration),
+
+    // Expressions
+    Identifier(&'a Identifier),
+
+    ArrayExpr(&'a ArrayExpr),
+    FunctionExpr(&'a FunctionExpr),
+    LogicalExpr(&'a LogicalExpr),
+    ObjectExpr(&'a ObjectExpr),
+    MemberExpr(&'a MemberExpr),
+    IndexExpr(&'a IndexExpr),
+    BinaryExpr(&'a BinaryExpr),
+    UnaryExpr(&'a UnaryExpr),
+    PipeExpr(&'a PipeExpr),
+    CallExpr(&'a CallExpr),
+    ConditionalExpr(&'a ConditionalExpr),
+    StringExpr(&'a StringExpr),
+    ParenExpr(&'a ParenExpr),
+
+    IntegerLit(&'a IntegerLit),
+    FloatLit(&'a FloatLit),
+    StringLit(&'a StringLit),
+    DurationLit(&'a DurationLit),
+    UintLit(&'a UintLit),
+    BooleanLit(&'a BooleanLit),
+    DateTimeLit(&'a DateTimeLit),
+    RegexpLit(&'a RegexpLit),
+    PipeLit(&'a PipeLit),
+
+    BadExpr(&'a BadExpr),
+
+    // Statements
+    ExprStmt(&'a ExprStmt),
+    OptionStmt(&'a OptionStmt),
+    ReturnStmt(&'a ReturnStmt),
+    BadStmt(&'a BadStmt),
+    TestStmt(&'a TestStmt),
+    BuiltinStmt(&'a BuiltinStmt),
+
+    // FunctionBlock
+    Block(&'a Block),
+
+    // Property
+    Property(&'a Property),
+
+    // StringExprPart
+    TextPart(&'a TextPart),
+    InterpolatedPart(&'a InterpolatedPart),
+
+    // Assignment
+    VariableAssgn(&'a VariableAssgn),
```

Comment:
Is `Assgn` an intentional spelling?  Should this be Assign?

Reply:
Its an intentional spelling of `Assignment`, open to suggestions on a better abbreviation on that one.

Reply:
I'm thinking `Assign`

when I role-play talking about it:

"It's a Variable Assign"